iroh-util/src/lock.rs

```diff
@@ -1,79 +1,304 @@
-use anyhow::Result;
-use file_guard::{FileGuard, Lock};
+use anyhow::{anyhow, Result as AnyhowResult};
 use std::fs::File;
+use std::io::prelude::*;
+use std::io::ErrorKind;
+use std::io::Write;
 use std::path::PathBuf;
-use std::rc::Rc;
+use sysinfo::{Pid, ProcessExt, ProcessStatus, System, SystemExt};
+use thiserror::Error;
+use tracing::warn;
 
-/// Manages a lock file used to track if an iroh program
-/// is already running.
-/// The lock is released either when the object is dropped
-/// or when the program stops.
+pub fn acquire_or_exit(lock: &mut ProgramLock) -> &mut ProgramLock {
```

Comment:
this should be a method on `ProgramLock`

Reply:
I am confused, why does this exist here and down below again?

Reply:
tried a few variations, and couldn't get the runtime to hang onto the reference at the call site:

what I wanted:
```rust
fn main() {
  let lock = ProgramLock::new("iroh-gateway")?.acquire_or_exit();
```
which caused the compiler to yell about an unused variable. changing `lock` -> `_lock` means the lock is `Drop`d right away, an makes the lock not work at all

Reply:
🤦 forgot to remove this

Reply:
> lock -> _lock means the lock is Dropd right away,

That is not true, this is how drop guards work, they are kept for the current scope. It is only dropped immediately if it is `_`.